src/freertos_drivers/esp_idf/EspIdfWiFi.cxx

```diff
@@ -0,0 +1,1409 @@
+                {
+                    // Cache the scan record for the user to read later.
+                    scanResults_.emplace_back((char*)ap_records[i].ssid,
+                        sec_type_encode(ap_records[i].authmode),
+                        ap_records[i].rssi);
+
+                    if (staIface_ && !connected_ && idx < 0)
+                    {
+                        int index = find_sta_profile(
+                            std::string((char*)ap_records[i].ssid));
+                        if (index >= 0 &&
+                            ap_records[i].authmode >= userCfg_.sta_[index].sec_)
+                        {
+                            // profile SSID and security mode match.
+                            idx = index;
+                            ssid = userCfg_.sta_[index].ssid_;
+                            pass = userCfg_.sta_[index].pass_;
+                            sec = userCfg_.sta_[index].sec_;
+                            conn_channel = ap_records[i].primary;
+                        }
+
+                    }
+                }
+            }
+            delete ap_records;
+            if (staIface_ && !connected_)
+            {
+                // If in STA mode and not connected, always be trying to make
+                // a connection.
+                if (idx < 0)
+                {
+                    // No profile match found, scan again.
+                    esp_wifi_scan_start(nullptr, false);
+                }
+                else
+                {
+                    // Profile match found, connect.
+                    sta_connect(ssid.c_str(), pass.c_str(), sec, conn_channel);
+                }
+            }
+            scan_finished();
+            break;
+        }
+        case WIFI_EVENT_AP_START:
+            ipAcquiredAp_ = true;
+            ip_acquired(IFACE_AP, true);
+            break;
+        case WIFI_EVENT_AP_STOP:
+            ipAcquiredAp_ = false;
+            ip_acquired(IFACE_AP, false);
+            break;
+        case WIFI_EVENT_AP_STACONNECTED:
+        {
+            wifi_event_ap_staconnected_t *event =
+                (wifi_event_ap_staconnected_t *)data;
+            LOG(INFO, "wifi: station " MACSTR " join, AID=%d",
+                MAC2STR(event->mac), event->aid);
+            ++apClientCount_;
+            break;
+        }
+        case WIFI_EVENT_AP_STADISCONNECTED:
+        {
+            wifi_event_ap_stadisconnected_t *event =
+                (wifi_event_ap_stadisconnected_t *)data;
+            LOG(INFO, "wifi: station " MACSTR " leave, AID=%d",
+                MAC2STR(event->mac), event->aid);
+            if (--apClientCount_ == 0)
+            {
+                ipLeased_ = false;
+            }
+            break;
+        }
+    }
+}
+
+//
+// EspIdfWiFi::ip_event_handler()
+//
+void EspIdfWiFi::ip_event_handler(esp_event_base_t base, int32_t id, void *data)
+{
+    if (base != IP_EVENT)
```

Comment:
Technically this should never happen, I wonder if conversion to `HASSERT(base == IP_EVENT);` would make sense?